src/device.rs

```diff
@@ -46,7 +46,7 @@ pub trait VirtioDevice<M: GuestAddressSpace>: Send {
         mem: M,
```

Comment:
It may make sense to remove the `mem` field from here, with the idea of passing it at construction time instead. `activate()` would forward or clone `self.mem` into the `Queue<M, C>` that it creates.

That would have the disadvantage of being an incompatible change, but it would be nice (and it would make sense) to remove the `GuestAddressSpace` argument from `VirtioDevice`. The `GuestAddressSpace` argument would still remain in the structs that implement the trait, e.g. `VirtioBlock`.

Reply:
The `VirtioDevice` interface (and implicitly, the signature of `activate`) has changed, but this RFC was not rebased on top of the latest version at the time.